tests/cpgrid/addLgrsOnDistributedGrid_test.cpp

```diff
@@ -397,9 +397,35 @@ BOOST_AUTO_TEST_CASE(threeLgrs)
     const std::array<double, 3> cell_sizes = {1.0, 1.0, 1.0};
     const std::array<int, 3> grid_dim = {10,8,8};
     grid.createCartesian(grid_dim, cell_sizes);
+
+    std::vector<int> parts(640);
+    for (int k = 0; k < 8; ++k) {
+        for (int j = 0; j < 8; ++j) {
+            for (int i = 0; i < 10; ++i)
+            {
+                const auto& elemIdx = (k*80) + (j*10) + i;
+                if (i<5) {
+                    if(j<4) { 
+                        parts[elemIdx] = 0;
+                    }
+                    if (j>3) {
+                        parts[elemIdx] = 1;
+                    }
+                }
+                if (i>4){
+                    if (j<4) {
+                        parts[elemIdx] = 2;
+                    }
+                    if (j>3){
+                        parts[elemIdx] = 3;
+                    }
```

Comment:
From a short glimpse:

Is there a particular reason why you do not use if-else? It seems like the second if condition always covers what a usual else branch would cover, too.

If-else might be easier to understand.

Reply:
no particular reason, I'll change it for else-if in coming commit